### PR TITLE
Project structure and simple CNN

### PR DESCRIPTION
**Pull request by mytkom**
Many things can be improved in code, but my main objective was to meet all the requirements for milestone 1.

I would be thankful for code review and local tests.

What I think should be made now:
1. ~~There are two additional metrics in the project description document, we should calculate and print them during training, validation and testing.~~ DONE
2. Early end of training in case of overfitting, there are many easy ways of doing it.
3. Evaluation on test dataset
4. Normalization of spectrograms in transform function
5. Choose lowest possible image size for sufficient CNN performance
6. ~~Maybe move this SpectrogramDataset class to ./src~~ DONE

Edited by Patonymous on 01.11.24:

I made a totally arbitrary call to format the code using `Black` formatter, but I'm open to choose any tool

### CODE REVIEW

File: src/training.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        if i % step == step - 1:
+            accuracy = correct / step
+            loss = running_loss / step
+            step = epoch * len(train_loader) + i
+            logger({"train/accuracy": accuracy, "train/loss": loss}, step=step)
+            running_loss = 0.0
+            correct = 0
+
+
+class metrics:
```
Comment (mytkom) [inline src/training.py]: I think that classes should start with upper case letter

Comment (mytkom) [reply src/training.py]: Btw great idea to return instance of a class with all these metrics!

**Review by mytkom — COMMENTED** [review]
Nice changes, thanks for fixing cuda execution, I forgot to check it on machine with GPU, so there were many places without `.to(device)` call.

**Comment by mytkom** [comment]
I propose to merge this PR, so it would be easier to review in the future.

@Willu12 and @bas0N can you take a look and review this PR?
